src/connection.ts

```diff
@@ -979,17 +980,59 @@ class Connection extends EventEmitter {
     this.curTransientRetryCount = 0;
     this.transientErrorLookup = new TransientErrorLookup();
 
-    this.state = this.STATE.CONNECTING;
-    this.state.enter!.call(this);
+    this.state = this.STATE.INITIALIZED;
+
+    process.nextTick(() => {
+      if (this.state === this.STATE.INITIALIZED) {
+        const message = 'In the next major version of `tedious`, creating a new ' +
+          '`Connection` instance will no longer establish a connection to the ' +
+          'server automatically. Please use the new `connect` helper function or ' +
+          'call the `.connect` method on the newly created `Connection` object to ' +
+          'silence this message.';
+        deprecate(message);
+        this.connect();
+      }
+    });
+  }
+
+  connect(connectListener?: (err?: Error) => void) {
+    if (this.state !== this.STATE.INITIALIZED) {
+      throw new ConnectionError('`.connect` can not be called on a Connection in `' + this.state.name + '` state.');
+    }
```

Comment:
This is something I'm not 100% sure about. `net.Socket` allows calling `.connect` multiple times (for whatever reason 🤷‍♂).

I'd argue that calling `.connect` more than once is a programmer error, and thus throwing an exception is the correct way to handle this.